src/js/player.js

```diff
@@ -1519,6 +1519,42 @@ class Player extends Component {
    * @private
    */
   handleTechClick_(event) {
+    const self = this;
+
+    // We want to handle single and double click events separately
+    // because they perform different actions.
+    // A single-click will pause/resume playback.
+    // A double-click will enter/exit fullscreen.
+
+    // If a subsequent click doesn't occur after a single click,
+    // within a short amount of time, trigger a single-click handler.
+    if (event.detail === 1) {
+      this.singleClickTimer_ = setTimeout(function() {
```

Comment:
this shouldn't be necessary because there's a native `dblclick` event we can use instead.

Reply:
I initially looked into utilising `dblclick`. However, the problem is that there's already `mousedown` listener on the tech which is triggered before `dblclick`. So using it would result in the player being paused/resumed when entering/exiting fullscreen.

Reply:
it should be fine because that should receive two events so it should pause and then unpause. I think it would still be better than trying to implement our own double click handler.

Reply:
Yep, that's what's happening — first click pauses, second click resumes playback — when entering/exiting fullscreen. While this is not a huge issue, it makes the experience somehow less seamless when toggling fullscreen. Firstly, the playback is visibly paused/resumed. The play toggle button also can be seen changing states. Furthermore,  in such cases unnecessary `play`, `pause` events are generated.

Reply:
So, looking at youtube and vimeo, they don't even up pausing and playing the video during double clicks and as far as I can tell they both use dblclick handlers. I think play toggle handling is just incorrect and needs to be re-worked because trying to implement it ourselves will just lead to various weird corner cases and issues. For example, I can get it to switch playback state and fullscreen state if I take a short while between the two clicks.

Reply:
The event order in `dblclick` is as follows: `mousedown -> mouseup -> click -> mousedown -> mouseup -> click -> dblclick`. If we look at re-working handling playback toggle and using native `dblclick`, we will still need some approach of cancelling the first click when `dblclick` occurs as having `click` — or in this particular case `mousedown` — and `dblclick` handler on the media element would trigger both handlers.